src/java/org/apache/cassandra/cql3/selection/ColumnFilterFactory.java

```diff
@@ -112,17 +115,19 @@ public ColumnFilter newInstance(List<Selector> selectors)
     {
         private final TableMetadata table;
         private final Set<ColumnMetadata> nonPKRestrictedColumns;
+        private final boolean returnStaticContentOnPartitionWithNoRows;
 
-        public OnRequestColumnFilterFactory(TableMetadata table, Set<ColumnMetadata> nonPKRestrictedColumns)
+        public OnRequestColumnFilterFactory(TableMetadata table, Set<ColumnMetadata> nonPKRestrictedColumns, boolean returnStaticContentOnPartitionWithNoRows)
```

Comment:
Nit: we can break this signature:
```suggestion
        public OnRequestColumnFilterFactory(TableMetadata table,
                                            Set<ColumnMetadata> nonPKRestrictedColumns,
                                            boolean returnStaticContentOnPartitionWithNoRows)
```